mautic/mautic
# Unable to perform "Change campaign" Action for removing a contact from one and adding to another campaign

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | 
| BC breaks?                             | 
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
Unable to perform "Change campaign" Action for removing a contact from one and adding to another campaign.


#### Steps to reproduce the bug:

1. Create a form FORM1
2. Create campaign B with 'source'-> FORM1 and 'action' -> anything
3. Create campaign A and go to builder of A.
4. Select 'contact source' -> FORM1 and 'action' -> 'Change Campaign'.
5. Go to that action and select 'Add to Campaign' -> 'B' and 'Remove from Campaign' -> "This campaign"
6. Publish both the campaign.
7. Go to the created form link and submit contact details.
8. Check in contact list of A and B.

**Expected :**
That newly created contact should be removed from A and added to B.

**Actual :**
That newly created contact is added to B but is not removed from A.

#### Steps to test this PR:

1. Create a form FORM1
2. Create campaign B with 'source'-> FORM1 and 'action' -> anything
3. Create campaign A and go to builder of A.
4. Select 'contact source' -> FORM1 and 'action' -> 'Change Campaign'.
5. Go to that action and select 'Add to Campaign' -> 'B' and 'Remove from Campaign' -> "This campaign"
6. Publish both the campaign.
7. Go to the created form link and submit contact details.
8. Check in contact list of A and B.
9. That newly created contact should be removed from A and added to B.
